src/analyze/analyze.c

```diff
@@ -188,14 +200,32 @@ static void free_unit_times(struct unit_times *t, unsigned n) {
         free(t);
 }
 
+static void subtract_timestamp(usec_t *a, usec_t b) {
+        if (*a > 0) {
+                assert(*a >= b);
+                if (*a < b) {
+                        log_warning("subtract_timestamp: cannot subtract %llu - %llu", (unsigned long long)*a, (unsigned long long)b);
+                        *a = 0;
+                } else
+                        *a -= b;
```

Comment:
I'm confused. If you assert that `a >= b`, the subsequent condition `a < b` will always be true and the actual subtraction will never take place. What am I missing?